index.rst

```diff
@@ -48,10 +48,205 @@
+
+* Flatfield screen monochrometer
+
+* Sky-spectrum monitor (if/when built)
+
+Like the data from the Auxillary Telescope, users want to interact with the LFA data through the butler, an appropriate dataset type having been define. Moreover users require these data with very low latencies as it is likely that they need it in order to make on-the-fly adjustments to systems during commissioning.
+
+We therefore propose that the LFA Announcement Stream is monitored by the DM-CS and when data from these enumerated sources is made available, that it be injected into the data backbone, from whence it will be treated like data (and not telemetry) by upstream services.
+
+
+Design-neutral Requirements
+===========================
+
+Rehardless of whether the ETL or new proposed architecture is adopted, the eventual architecture needs to show how it can meet satisfy the following requirments and use cases.  
+
+
+Availability of the DM-EFD capabilities
+----------------------------------------
+
+If, as anticipated, DM tooling is the primary of interface to EFD data for anyone beyond hardware-level engineers, availability of those services will be important to operational staff in Chile and the US, as well as to science users. It is therefore a requirement that the entire architecture is structured so that sandbox deployments, rolling upgrades and carefully coordinated downtime are the norm for routine operations. 
+
+Interfaces
+----------
+
+Data should be available via TAP/ADQL services as other data sources available to the Science Platform.
+
+The interface to the Science Platform should be deployment- and time-invariant: the same notebook accessing EFD data should run without modifications on the day in Chile and a month later at the LDF.
+
+A syntactic sugar to make access to EFD data more pythonic from the notebook (and to shield the user from schema implementation details) has been requested. Here is an example of how a notebook user could obtain statistics on the M1/M3 temperature sensors::
+
+  import numpy as np
+  import lsst.efd as efd
+  ...
+
+  # Get the temperatures in one go 
+  envtemp = efd.get("m1m3.actuators.envtemp")
+  stdev = np.std(envtemp)
+  mean = np.mean(envtemp)
+  print(f"temperature = {mean} +/- {stdev} K")
+
+
+
+Aggregation
+-----------
+
+The purpose of aggregation is both to reduce volume on high-frequency telemetry data and to increase the signal-to-noise of busy telemetry. Science Platform users are generally interested in events at the same order of cadence as a camera exposure; therefore we propose that all telemetry data sampled with a frequency higher than 1Hz is (1) sampled at 1Hz and (2) aggregated to 1Hz using these generic statistics:
+
+* Max
+
+* Min
+
+* Mean
+
+* Median
+
+* Standard Deviation
+
+For command streams, no aggregation should be done.
+
+For event streams we propose that using Kafka we sum repeats of the
```

Comment:
Can Kafka deal with some fields in the event changing between "repeats"?  We would have to identify which fields those are, of course.

Reply:
I don't believe we have looked into Kafka in enough detail to answer low level technical questions.  If this high level design is accepted, we'll have to do the detailed technical review of the possible underlying technologies.